embuilder.py

```diff
@@ -76,68 +101,60 @@ def build(src, result_libs, args=[]):
     for lib in result_libs:
       assert os.path.exists(shared.Cache.get_path(lib)), 'not seeing that requested library %s has been built because file %s does not exist' % (lib, shared.Cache.get_path(lib))
 
+
 def build_port(port_name, lib_name, params):
-  build('''
-    int main() {}
-  ''', [os.path.join('ports-builds', port_name, lib_name)] if lib_name else None, params)
+  build(C_BARE, [os.path.join('ports-builds', port_name, lib_name)] if lib_name else None, params)
+
+
+SYSTEM_TASKS = ['compiler-rt', 'libc', 'libc-mt', 'dlmalloc', 'dlmalloc_threadsafe', 'pthreads', 'dlmalloc_debug', 'libcxx', 'libcxx_noexcept', 'libcxxabi', 'html5']
+USER_TASKS = ['al', 'gl', 'binaryen', 'bullet', 'freetype', 'libpng', 'ogg', 'sdl2', 'sdl2-image', 'sdl2-ttf', 'sdl2-net', 'vorbis', 'zlib']
 
 operation = sys.argv[1]
 
 if operation == 'build':
+  auto_tasks = False
   tasks = sys.argv[2:]
+  if 'SYSTEM' in tasks:
+    tasks = SYSTEM_TASKS
```

Comment:
Looks like this adds a new 'SYSTEM' option? Good idea, but some notes:

1. Needs to be documented in embuilder's help.
2. Currently, `zlib SYSTEM` will just build system, ignoring the user-passed zlib. Perhaps it should be both?

Reply:
Yeah, maybe we could have user and system both, and the args mechanism could probably be improved. Although which libs are in that category are a bit open to interpretation. I'd probably consider HTML5 and AL to be user rather than system, but it's In terms of mechanisms, we have them as auto-built libs like the system libs rather than ports. I'm guessing that's just a historical thing?

Reply:
Partly historical, yeah. What is bundled in emscripten is a system lib, and stuff that we fetch on demand is in ports.

Oh, also, we should add a test in `sanity.test_embuilder` for the new stuff in this PR.

Reply:
Oh ok; maybe we should just call the options `BUILTIN` and `PORTS` then.